src/callbacks_stage/subcell_bounds_check_2d.jl

```diff
@@ -49,40 +39,35 @@
                 continue
             end
             key = Symbol(string(v), "_min")
-            deviation_threaded = idp_bounds_delta_local[key]
-            @threaded for element in eachelement(solver, cache)
-                deviation = deviation_threaded[stride_size * Threads.threadid()]
+            deviation = idp_bounds_delta_local[key]
+            @batch reduction=(max, deviation) for element in eachelement(solver, cache)
```

Comment:
Comment on `@batch`, please

Reply:
Do you mean a comment about why we're using `@batch` and not `@threaded` or an explanation on how `@batch reduction` works/what it does?
For now, I added a comment about the first one at the top of this routine.

Reply:
Just a general comment why we're using `@batch` here, xref https://github.com/trixi-framework/Trixi.jl/pull/1888#discussion_r1537932975

Reply:
Done in [51ad911](https://github.com/trixi-framework/Trixi.jl/pull/1888/commits/51ad911282dcc1f4704ee7b7a973fb98bcf7beba)